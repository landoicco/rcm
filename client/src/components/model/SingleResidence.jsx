//@flow

import React from "react";
import type { DataItem } from "../../commons/types";

const SingleResidence = ({ item }: DataItem): React.Node => {
  return (
    <div>
      <div className="card teal accent-4">
        <div className="card-content">
          <span className="card-title">
            {item.street} St., {item.extNumber}
          </span>
        </div>
        <div className="card-action">
          <p>Street: {item.street}</p>
          <p>Exterior number: {item.extNumber}</p>
        </div>
      </div>
    </div>
  );
};

export default SingleResidence;
