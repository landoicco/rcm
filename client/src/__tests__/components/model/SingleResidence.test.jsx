import { render, screen } from "@testing-library/react";
import SingleResidence from "../../../components/model/SingleResidence";

const mockResidence = {
  id: "-1",
  street: "here",
  extNumber: "000",
  address: "here 000",
};

describe("SingleResidence", () => {
  test("The component render correctly", () => {
    const { container } = render(<SingleResidence item={mockResidence} />);
    expect(container).toMatchSnapshot();
  });
});
