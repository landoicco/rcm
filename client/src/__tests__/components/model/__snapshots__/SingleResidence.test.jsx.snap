// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleResidence The component render correctly 1`] = `
<div>
  <div>
    <div
      class="card teal accent-4"
    >
      <div
        class="card-content"
      >
        <span
          class="card-title"
        >
          here
           St., 
          000
        </span>
      </div>
      <div
        class="card-action"
      >
        <p>
          Street: 
          here
        </p>
        <p>
          Exterior number: 
          000
        </p>
      </div>
    </div>
  </div>
</div>
`;
